code/modules/reagents/Chemistry-Machinery.dm

```diff
@@ -1414,3 +1415,356 @@ USE THIS CHEMISTRY DISPENSER FOR MAPS SO THEY START AT 100 ENERGY
+		/obj/item/weapon/reagent_containers/food/snacks/grown/tomato = list("tomatojuice",0),
+		/obj/item/weapon/reagent_containers/food/snacks/grown/carrot = list("carrotjuice",0),
+		/obj/item/weapon/reagent_containers/food/snacks/grown/berries = list("berryjuice",0),
+		/obj/item/weapon/reagent_containers/food/snacks/grown/banana = list("banana",0),
+		/obj/item/weapon/reagent_containers/food/snacks/grown/potato = list("potato",0),
+		/obj/item/weapon/reagent_containers/food/snacks/grown/lemon = list("lemonjuice",0),
+		/obj/item/weapon/reagent_containers/food/snacks/grown/orange = list("orangejuice",0),
+		/obj/item/weapon/reagent_containers/food/snacks/grown/lime = list("limejuice",0),
+		/obj/item/weapon/reagent_containers/food/snacks/watermelonslice = list("watermelonjuice",0),
+		/obj/item/weapon/reagent_containers/food/snacks/grown/poisonberries = list("poisonberryjuice",0),
+	)
+
+
+	var/obj/item/crushable = null
+
+
+/obj/item/weapon/reagent_containers/mortar/afterattack(obj/target, mob/user , flag)
+	if(target.is_open_container() && target.reagents) //Something like a glass. Player probably wants to transfer TO it.
+		if(!reagents.total_volume)
+			user << "<span class='warning'>[src] is empty.</span>"
+			return
+
+		if(target.reagents.total_volume >= target.reagents.maximum_volume)
+			user << "<span class='warning'>[target] is full.</span>"
+			return
+
+		var/trans = src.reagents.trans_to(target, amount_per_transfer_from_this)
+		user << "<span class='notice'>You transfer [trans] units of the solution to [target].</span>"
+
+/obj/item/weapon/reagent_containers/mortar/attackby(var/obj/item/O as obj, var/mob/user as mob)
+	if (isscrewdriver(O))
+		if(crushable)
+			crushable.loc = src.loc
+		new /obj/item/stack/sheet/metal(user.loc)
+		new /obj/item/trash/bowl(user.loc)
+		return
+	if (crushable)
+		user << "<span class ='warning'>There's already something inside!</span>"
+		return 1
+	if (!is_type_in_list(O, blend_items) && !is_type_in_list(O, juice_items))
+		user << "<span class ='warning'>You can't grind that!</span>"
+		return ..()
+	if(istype(O, /obj/item/stack/sheet))
+		var/obj/item/stack/sheet/metal/M = new /obj/item/stack/sheet/metal(src)
+		var/obj/item/stack/sheet/S = O
+		S.use(1)
+		crushable = M
+		return 0
+	user.drop_item(O, src)
+	crushable = O
+	return 0
+
+/obj/item/weapon/reagent_containers/mortar/attack_hand(mob/user as mob)
+	if(user.get_inactive_hand() != src) return ..()
+	if(crushable)
+		crushable.loc = src.loc
+		user.put_in_active_hand(crushable)
+		crushable = null
+	return
+
+/obj/item/weapon/reagent_containers/mortar/attack_self(mob/user as mob)
+	if(!crushable)
+		user << "<span class='notice'>There is nothing to be crushed.</span>"
+		return
+	if (reagents.total_volume >= volume)
+		user << "<span class='warning'>There is no more space inside!</span>"
+		return
+	if(is_type_in_list(crushable, juice_items))
+		user << "<span class='notice'>You smash the contents into juice!</span>"

```

Comment:
Consider adding a special branching option if in both lists, to allow you your choice of juice or dust.